src/biokbase/narrative/jobs/jobcomm.py

```diff
@@ -514,6 +515,14 @@ def send_comm_message(self, msg_type: str, content: dict) -> None:
         Sends a ipykernel.Comm message to the KBaseJobs channel with the given msg_type
         and content. These just get encoded into the message itself.
         """
+        # For STATUS responses, add a last_checked field
+        # to each output_state. Note: error states will have
+        # the last_checked field too
+        if msg_type == MESSAGE_TYPE["STATUS"]:
+            now = time_ns()
+            for output_state in content.values():
+                output_state["last_checked"] = now
```

Comment:
why not add this timestamp when the job manager is putting together the list of jobs, instead of adding an extra iteration through the job state data here?

Reply:
Wasn't sure since the CANCEL_JOBS request also responds with a STATUS message.

Reply:
I decided not to filter the STATUS response for CANCEL_JOBS though because I figured in theory they should all get updated, whether successfully or just coming back with an error

Reply:
Because everything is asynchronous, the FE doesn't have any way of knowing what triggered a job status message -- whether it was a cancel request, a status request, or the BE job loop. That's why I say it's better to put the timestamp on in the job manager, so that all job state objects that the FE receives have a timestamp on them.

Reply:
I think one allure of putting everything into JobComm is less tests surgery ... But putting it deep into the stack, at the origin of the STATUS response ds, seems less googly-eyed

Reply:
Okay I tried putting all the filtering/last_checked logic at the source `_construct_job_state_set` but the tests were complaining so I'm abandoning that effort for the sake of time. Is the current placement of the filtering/last_checked good enough?